src/core/control/PdfCache.cpp

```diff
@@ -75,15 +78,15 @@ PdfCacheEntry* PdfCache::cache(XojPdfPageSPtr popplerPage, cairo_surface_t* img,
     return ne;
 }
 
-void PdfCache::render(cairo_t* cr, const XojPdfPageSPtr& popplerPage, double zoom) {
-    this->renderMutex.lock();
+void PdfCache::render(cairo_t* cr, size_t pageNo, double zoom, double pageWidth, double pageHeight) {
+    std::lock_guard<std::mutex> lock(this->renderMutex);
 
     this->setZoom(zoom);
 
-    PdfCacheEntry* cacheResult = lookup(popplerPage);
+    PdfCacheEntry* cacheResult = lookup(pageNo);
     bool needsRefresh{cacheResult == nullptr};
```

Comment:
Random unrelated nitpick, but the brace initialization syntax here makes this unnecessarily hard to read. Could you fix this to use `=` ?